packages/swingset-runner/src/kerneldump.js

```diff
@@ -0,0 +1,193 @@
+import path from 'path';
+import process from 'process';
+
+import { openSwingStore } from '@agoric/swing-store-lmdb';
+
+function usage() {
+  console.log(`
+Command line:
+  kerneldump [FLAGS...] [BASEDIR]
+
+FLAGS may be:
+  --raw  - just dump the kernel state database as key/value pairs alphabetically without annotation
+  --help - print this helpful usage information
+
+BASEDIR is the base directory where a swingset's vats live
+  If BASEDIR is omitted it defaults to the current working directory.
+`);
+}
+
+function fail(message, printUsage) {
+  console.log(message);
+  if (printUsage) {
+    usage();
+  }
+  process.exit(1);
+}
+
+function pkv(key, value) {
+  console.log(`${key} :: ${value}`);
+}
+
+function p(str) {
+  console.log(str);
+}
+
+function gap() {
+  p('');
+}
+
+/* eslint-disable no-use-before-define */
+export function main() {
+  const argv = process.argv.splice(2);
+  let rawMode = false;
+  while (argv[0] && argv[0].startsWith('-')) {
+    const flag = argv.shift();
+    switch (flag) {
+      case '--raw':
+        rawMode = true;
+        break;
+      case '--help':
+        usage();
+        process.exit(0);
+        break;
+      default:
+        fail(`invalid flag ${flag}`, true);
+        break;
+    }
+  }
+
+  const basedir = argv.shift();
+  const kernelStateDBDir = path.join(basedir, 'swingset-kernel-state');
```

Comment:
I'd want to use this tool to examine the two swingsets created by `agoric start`, one of which uses `.../_agstate/agoric-servers/dev/fake-chain-fake-chain-state/data.mdb` and the other is `.../_agstate/agoric-servers/dev/swingset-kernel-state/data.mdb`. Maybe the `argv` should just be used as `kernelStateDBDir` without appending a `swingset-kernel-state` suffix? Or maybe if the `argv` path ends in `data.mdb`, then chop off enough path components to turn it into something that can be passed into `openSwingStore` ?

Reply:
The command line UI was intended to mimic `bin/runner` in how you specify the swingset of interest, without requiring the user to understand the internal details of how the database files are organized, but `agoric start` has two swingsets in the same directory so at least one of them has to follow a different convention for where it keeps its state.  Hosts that organize their database files in a different way are a use case that needs to be accommodated.  I'm generally wary of trying to put too much heuristic squishiness into tooling, but in this case perhaps probing the path you give it for either a `data.mdb` file or a `swingset-kernel-state` directory might make sense.